src/arrayMethodJoin.js

```diff
@@ -5,7 +5,16 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = separator === undefined
+      ? ',' : separator === null ? 'null' : separator.toString();
```

Comment:
Try to do a simple check, what happens, if you: 

```js
const exapmleArray = ['hello', 'world']
console.log(exampleArray.join(null))
console.log(exampleArray.join2(null))
```

You should not treat `null` differently than `undefined` here

Reply:
i just changed to:
const separator = String(sep);